### PR TITLE
Add export sync

### PR DESCRIPTION
**Pull request by MrAlias**
The batching log processor will generate records from 4 different locations (polling, OnEmit, ForceFlush, Shutdown). In order to ensure an Exporter is called serially, as is required by the interface, this function will be used in the processor to centralize the interaction with its Exporter.

Part of #5063.

See #5093 for the implementation use.

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — COMMENTED** [review]
Just curious, why did you choose this approach, rather than just locking around the exporter?

**Review by XSAM — APPROVED** [review]
👍

**Comment by MrAlias** [comment]
> Just curious, why did you choose this approach, rather than just locking around the exporter?

Locking around the exporter would mean the calling goroutine would need to wait for the export to complete before continuing. Either the batching processor routine, or the polling routine will be blocked in that case. Either of which is going to be problematic as it could cause large back-pressure for systems with high throughput.

Passing via a channel to a separate goroutine will allow the the export to not block, and if the export becomes excessively slow we can start dropping payloads instead of locking up.

File: sdk/log/exporter.go [hunk 1]
```diff
@@ -50,3 +52,51 @@ func (noopExporter) Export(context.Context, []Record) error { return nil }
...
+	// respCh is the channel any error returned from the export will be sent
+	// on. If this is nil, and the export error is non-nil, the error will
+	// passed to the OTel error handler.
+	respCh chan<- error
+}
+
+// DoExport calls exportFn with the data contained in e. The error response
+// will be returned on e's respCh if not nil. The error will be handled by the
+// default OTel error handle if it is not nil and respCh is nil or full.
+func (e exportData) DoExport(exportFn func(context.Context, []Record) error) {
```
Comment (dashpole) [inline sdk/log/exporter.go:84]: does this need to be exported?

**Review by dashpole — APPROVED** [review]

File: sdk/log/exporter.go (hunk 1 above)
Comment (dashpole) [reply sdk/log/exporter.go:84]: Its just a personal preference.  I'm OK with this as written.

**Pull request merged**